app/controllers/wiki_pages_api_controller.rb

```diff
@@ -616,7 +616,9 @@ def get_update_params(allowed_fields=Set[])
     # needs to be moved to the controller where we know who is
     # editing it.
     if page_params.include?(:body)
-      unless BeyondZConfiguration.unrestricted_html_users.include?(@current_user.id)
+      # unless BeyondZConfiguration.unrestricted_html_users.include?(@current_user.id)
+      # This checks if the current user is a teacher of the course this page is attached to - replaces the old security whitelist.
+      unless @current_user && @context && @context.respond_to?(:grants_any_right?) && can_do(@context, @current_user, :manage)
```

Comment:
On Thu, Jul 09, 2015 at 09:26:21AM -0700, Brian Sadler wrote:
> Wow, that is a nutty way to check roles.  It's amazing that all these "does this object exist" things aren't abstracted away in a helper or something.  Anyway, I think this is how I've seen it in the codebase.

Aye, I borrowed this from the view which needs to be over-defensive
in case it is called on something like the login page, where none
of that exists. I think a wiki edit is always going to have those
objects... but I'm not 100% sure about that, so better to check
it and run the filter if they are null than risk some guest user
being able to inject code.